game-core/src/main/java/games/strategy/engine/data/ModifiableProperty.java

```diff
@@ -0,0 +1,145 @@
+      public void setValue(String string) throws Exception {
+        stringSetter.accept(string);
+      }
+
+      @Override
+      public void setValue(T value) throws Exception {
+        setter.accept(value);
+      }
+
+      @Override
+      public T getValue() {
+        return getter.get();
+      }
+
+      @Override
+      public void resetValue() {
+        resetter.run();
+      }
+    };
+  }
+
+  /**
+   * Convenience method to create an instance of this interface with no resetter.
+   */
+  static <T> ModifiableProperty<T> of(
+      final ExceptionConsumer<T> setter,
+      final ExceptionConsumer<String> stringSetter,
+      final Supplier<T> getter) {
+    return of(setter, stringSetter, getter, () -> throwIllegalStateException("No Resetter"));
+  }
+
+  /**
+   * Convenience method to create a generic String instance of this interface.
+   */
+  static ModifiableProperty<String> of(
+      final ExceptionConsumer<String> setter,
+      final Supplier<String> getter,
+      final Runnable resetter) {
+    return of(setter, setter, getter, resetter);
+  }
+
+  /**
+   * Convenience method to create an instance of this interface that only gets.
+   */
+  static <T> ModifiableProperty<T> of(final Supplier<T> getter) {
+    return of(
+        t -> throwIllegalStateException("No Setter"),
+        t -> throwIllegalStateException("No String Setter"),
+        getter,
+        () -> throwIllegalStateException("No Resetter"));
+  }
+
+  /**
+   * Convenience method to create an instance of this interface that only sets, but doesn't reset.
+   */
+  static <T> ModifiableProperty<T> of(
+      final ExceptionConsumer<T> setter,
+      final ExceptionConsumer<String> stringSetter) {
+    return of(
+        setter,
+        stringSetter,
+        () -> throwIllegalStateException("No Getter"),
+        () -> throwIllegalStateException("No Resetter"));
+  }
+
+
+  /**
+   * Convenience method to create an instance of this interface that just contains a direct
+   * setter and getter. And no support for Strings as secondary setter.
+   */
+  static <T> ModifiableProperty<T> ofSimple(
+      final ExceptionConsumer<T> setter,
+      final Supplier<T> getter) {
+    return of(setter,
+        o -> throwIllegalStateException("No String Setter"),
+        getter,
+        () -> throwIllegalStateException("No Resetter"));
+  }
+
+
+  /**
+   * Convenience method to create an instance of this interface that just contains a
+   * getter. And no support for setters of any kind.
+   */
+  static <T> ModifiableProperty<T> ofSimple(final Supplier<T> getter) {
+    return ofSimple(e -> throwIllegalStateException("No Setter"), getter);
+  }
+
+  static <T> T throwIllegalStateException(final String text) {
+    throw new IllegalStateException(text);
+  }
+
+  /**
+   * A Consumer capable of throwing a GameParseException.
+   *
+   * @param <T> The type of Object to consume.
+   */
+  @FunctionalInterface
+  static interface ExceptionConsumer<T> {
```

Comment:
* Recommend naming this `ThrowingConsumer` and parameterize the type of exception thrown.  Take a look at `BackgroundTaskRunner$ThrowingSupplier`.
* `static` is superfluous on an interface nested within another interface.
* The Javadocs still reference `GameParseException`.